Generator/ProductGenerator.php

```diff
@@ -159,6 +172,13 @@ public function generate(array $config, $outputDir, ProgressHelper $progress, ar
             $this->forcedValues = [];
         }
 
+        $this->variantGroupCounts = [];
+        if ($variantGroupCount > 0) {
+            foreach ($this->groupRepository->getAllVariantGroups() as $variantGroup) {
```

Comment:
idea is to generate 100k variant groups, maybe fetch only the codes could be better (if doable) ;)